src/com/facebook/buck/go/GoBuckConfig.java

```diff
@@ -109,7 +110,11 @@ Path getDefaultPackageName(BuildTarget target) {
       public Tool get() {
         Optional<Path> toolPath = delegate.getPath("go", configName);
         if (!toolPath.isPresent()) {
-          toolPath = Optional.of(goToolDirSupplier.get().resolve(toolName));
+          String exeName = toolName;
+          if (delegate.getPlatform() == Platform.WINDOWS) {
+            exeName = toolName + ".exe";
+          }
```

Comment:
`ExecutableFinder`'s constructor lets you pass in the env, so if you know where it should be, you can do something like this:
```
toolPath = new ExecutableFinder().getOptionalExectuable(
    Paths.get(toolname),
    goToolDirSupplier.get());
```

This has the added advantage of making sure it's executable on unix systems.  But I also agree that all this can be done in a different PR.